### DISCUSSION TITLE
Define datacube parameters correctly

### DISCUSSION THREAD

**Discussion by dkczk** [davidfrantz/force#92] [Q&A]
Hey guys,

I'm having trouble to define the datacube parameters correctly. Especially the parameters TILE_SIZE, BLOCK_SIZE, ORIGIN_LON, ORIGIN_LAT. The [documentation](https://force-eo.readthedocs.io/en/latest/howto/datacube.html) says: 

- ORIGIN_LAT / ORIGIN_LON
> ORIGIN_LAT and ORIGIN_LON are the origin coordinates of the grid system in decimal degree. The upper left corner of tile X0000_Y0000 represents this point. It is a good choice to use a coordinate that is North-West of your study area – to avoid negative tile numbers.

Let's say I want to use the [GLANCE7-Grid for europe](https://measures-glance.github.io/glance-grids/grids/eu) and I have a study area in eastern Germany. If I want a correctly tiled grid, I'd need to define the grid systems origin which is somewhere in the north east of Pennsylvania. How do you get this coordinate? I couldn't find anything but the [coordinates of the grids center](https://measures-glance.github.io/glance-grids/params). 

- TILE_SIZE
> TILE_SIZE is the tile size (in target units, commonly in meters). Tiles are square.

For the GLANCE7-grid it would mean a TILE_SIZE of 150000 as one pixel has an edge length of 30 meters. Am I correct here? Furthermore if I'd plan to use Landsat-8 and Sentinel-2 data and want to have both of them in a harmonized grid, would RESOLUTION_LANDSAT = 10 resample the Landsat data to Sentinel-2s 10 meters per pixel edge length? Therefore I'd need to adjust the TILE_SIZE to 450000, is that correct?

- BLOCK_SIZE
> BLOCK_SIZE is the block size (in target units, commonly in meters) of the image chips. Blocks are stripes, i.e. they are as wide as the tile and as high as specified here. The blocks represent the internal structure of the GeoTiffs, and represent the primary processing unit of the force-higher-level routines.

In my example one block of the „harmonized" GLANCE7-Grid would be three times the tile size so 1350000?

Sorry for asking so much basic questions but I ended up with false gridded Level-2 Landsat-8 images and the tutorial couldn't answer the questions above for me.

Have a nice day and stay healthy,

Cheers

**Comment by davidfrantz** [comment]
Dear @dkczk ,

GLANCE7 is a predefined grid, which was built-in so that you don't have to worry about all these issues ;)

The tutorial says: 
`Two default projection / grid systems are predefined in FORCE. They can be specified via the PROJECTION parameter instead of giving a WKT string. The predefined options have their own settings for ORIGIN_LAT, ORIGIN_LON, TILE_SIZE, and BLOCK_SIZE, thus the values given in the parameterfile will be ignored.`

The 150km grid can shelter 30m Landsat as well as 10m Sentinel-2.

This should work out of the box ;)
Cheers,
David

**Comment by davidfrantz** [comment]
No, this only applies if you define a grid yourself, i.e. if you specify your own WKT string.

GLANCE7 is predefined. Simply give:

``PROJECTION = GLANCE7`` and ignore all other cubing parameters.

As the tutorial says, the predefined grids come with their own settings.
`Two default projection / grid systems are predefined in FORCE. They can be specified via the PROJECTION parameter instead of giving a WKT string. The predefined options have their own settings for ORIGIN_LAT, ORIGIN_LON, TILE_SIZE, and BLOCK_SIZE, thus the values given in the parameterfile will be ignored.`

Cheers

**Reply by dkczk** [reply]
Bummer. I was confused by the definitions section of your tutorial. Thank you very much by your quick help.
